firmware/port.c

```diff
@@ -126,7 +128,7 @@ void port_disable(PortData* p) {
 
 void port_send_status(PortData* p, u8 d) {
     if (p->reply_len >= BRIDGE_BUF_SIZE) {
-        invalid();
+        bridge_disable_chan(p->chan);
```

Comment:
:+1:  `invalid()` wasn't supposed to live past when I was running this with the debugger attached at all times.

Once you have this otherwise working, try adding a call to `bridge_enable_chan` in `port_disable` to let it semi-gracefully recover from errors.

The current code here will leave the channel permanently disabled after one of these error conditions, which is at least better than hanging the MCU, as `invalid()` does without the debugger attached. `port_disabled` is called when the port is closed by the SoC ("opened" = false in the terminology of my previous comment), so after the connection causing the error  is dropped and the state is reset, the port will accept new connections.

1. client puts the port in an invalid state
2. this file detects the error and calls `bridge_disable_chan()`
3. samd21 sends a packet with `enabled` = false
4. spid receives the packet and drops the connection
5. spid replies with `opened` = false, because there is no connection
6. samd21 calls `port_disable`, which resets the port state and sets `enabled = true` on the channel
7. spid can accept a new connection on that port